web/react/components/sidebar.jsx

```diff
@@ -197,6 +197,19 @@ module.exports = React.createClass({
             if (UserStore.getCurrentId() === msg.user_id) {
                 AsyncClient.getChannels(true);
             }
+        } else if(msg.action === "user_removed") {
+            if(msg.user_id === UserStore.getCurrentId()) {
+                AsyncClient.getChannels(true);
+
+                if(msg.props.channel_id === ChannelStore.getCurrentId() && $('#removed_from_channel').length > 0) {
+                    var channelName = ChannelStore.getCurrent().display_name;
+                    var curUser = UserStore.getProfile(msg.props.remover).username;
+                    $('#removed_from_channel').find('.modal-title').text("Removed from " + channelName);
+                    $('#removed_from_channel').find('.modal-body').children().text(curUser + " removed you from " + channelName);
+
+                    $('#removed_from_channel').modal('show');
```

Comment:
Normally, we would pass arguments through the data-* attributes of the target element that is getting clicked on, but since nothing is getting clicked that's not an option. Bootstrap doesn't really provide a guideline on how to handle this when the modal is manually triggered like we are doing here. I think using the AppDispatcher would be overkill for this. I think we already have a case like this in edit_post_modal.jsx that opens up delete_post_modal.jsx and uses the BrowserStore to transfer state. I think that's the best way to go about it without dispatching an event or directly manipulating the contents of a component.